Tools/autotest/arducopter.py

```diff
@@ -1603,6 +1603,58 @@ def MaxAltFence(self):
 
         self.zero_throttle()
 
+    # MaxAltFence - fly up and make sure fence action does not trigger
+    # Also check that the vehicle will not try and descend too fast when trying to backup from a max alt fence due to avoidance
+    def MaxAltFenceAvoid(self):
+        '''Test Max Alt Fence Avoidance'''
+        self.takeoff(10, mode="LOITER")
+        """Hold loiter position."""
+
+        # enable fence, only max altitude, defualt is 100m
+        # No action, rely on avoidance to prevent the breach
+        self.set_parameters({
+            "FENCE_ENABLE": 1,
+            "FENCE_TYPE": 1,
+            "FENCE_ACTION": 0,
+        })
+
+        # Try and fly past the fence
+        self.set_rc(3, 1920)
+
+        # Avoid should prevent the vehicle flying past the fence, so the altitude wait should timeouts
+        try:
+            self.wait_altitude(140, 150, timeout=90, relative=True)
```

Comment:
```suggestion
            self.wait_altitude(140, 15000, timeout=90, relative=True)
```

removes the race condition

You could also simplify this to simply ensure you remain just below the fence instead, removing the try/except/except block

Reply:
Sorry, I don't understand. Increasing the upper bound would mean it always times out? But the test wants it to time out?

Reply:
peter is happy with it